pkg/server/socket.go

```diff
@@ -31,31 +30,8 @@ const (
 // FilesystemPreRequirements ensures the target `rundir` features the correct
 // permissions.
 func FilesystemPreRequirements(rundir string) error {
-	socketpath := SocketPath(rundir)
 	if err := os.RemoveAll(rundir); err != nil && !os.IsNotExist(err) {
-		info, err := os.Stat(rundir)
-		if err != nil {
-			return fmt.Errorf("failed to stat old pod info socket directory %s: %v", rundir, err)
-		}
-		// Owner must be root
-		tmp := info.Sys()
-		statt, ok := tmp.(*syscall.Stat_t)
-		if !ok {
-			return fmt.Errorf("failed to read pod info socket directory stat info: %T", tmp)
-		}
-		if statt.Uid != 0 {
-			return fmt.Errorf("insecure owner of pod info socket directory %s: %v", rundir, statt.Uid)
-		}
-
-		// Check permissions
-		if info.Mode()&fullReadWriteExecutePermissions != thickPluginSocketRunDirPermissions {
-			return fmt.Errorf("insecure permissions on pod info socket directory %s: %v", rundir, info.Mode())
-		}
-
-		// Finally remove the socket file so we can re-create it
-		if err := os.Remove(socketpath); err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("failed to remove old pod info socket %s: %v", socketpath, err)
-		}
+		return fmt.Errorf("failed to remove old pod info socket directory %s: %v", rundir, err)
```

Comment:
nit: If we're removing all the failure cases, we could improve the message to say `failed to remove old pod info socket directory (please stat the socketfile and check the mode is restrictive and UID != 0)`